Ginger/Ginger/Variables/AddVariablePage.xaml

```diff
@@ -21,7 +24,41 @@
                         </DockPanel>
                     </Border>
                 </TabItem.Header>
-                <uclistview:UcListView x:Name="xLibraryTabListView" ListTitleVisibility="Collapsed" ListSelectionMode="Extended"/>
+
+                <StackPanel Orientation="Vertical">
+                    <DockPanel Grid.Row="0" HorizontalAlignment="Stretch" Name="xVariableDetailsDockPanel"  Margin="10,25,0,10">
+                        <StackPanel Orientation="Horizontal" DockPanel.Dock="Top">
+                            <Label Style="{StaticResource $LabelStyle}" MinWidth="70" Content="Name:"/>
+                            <StackPanel Orientation="Vertical">
+                                <TextBox Width="250" Height="25" Style="{StaticResource $TextBoxStyle}" Name="variableName" />
+                                <TextBlock Text="Name is mandatory" Name="NameError" DockPanel.Dock="Bottom" Foreground="Red" Visibility="Hidden"/>
+                            </StackPanel>
+                        </StackPanel>
+
+
+                        <StackPanel Orientation="Horizontal" DockPanel.Dock="Top">
+                            <StackPanel Orientation="Horizontal">
+                                <Label Style="{StaticResource $LabelStyle}" MinWidth="50" Content="Description:"/>
+                                <TextBox Height="25" Width="250"  Style="{StaticResource $TextBoxStyle}"  Name="variableDescription"/>
+                            </StackPanel>
+
+                            <StackPanel Orientation="Horizontal" Name="ValueStackPanel" Visibility="Collapsed" Margin="5, 0, 0, 0">
+                                <Label Style="{StaticResource $LabelStyle}" MinWidth="50" Content="Initial Value:"/>
+                                 <TextBox  Height="25" Width="250" Style="{StaticResource $TextBoxStyle}"  Name="variableValue"/>
+                                 <TextBlock Text="Value is mandatory" Name="ValueError" DockPanel.Dock="Bottom" Foreground="Red" Visibility="Hidden" Margin="5, 0, 0 , 0"/>
```

Comment:
Add controls for Marking variable as input, output, mandatory and publish

Reply:
Added the required controls panel